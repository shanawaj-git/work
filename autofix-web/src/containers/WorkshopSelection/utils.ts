import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
