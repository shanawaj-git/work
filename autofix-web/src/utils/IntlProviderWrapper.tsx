import React, { useMemo } from "react";
import { getTranslation, TranslationsType } from "../translations";
import { IntlProvider } from "react-intl";
import { flatData, I18NLocale } from ".";

export const withWrapper = (component: JSX.Element, pageProps = { en: {} }) => (
  <IntlProviderWrapper locale="en" pageProps={pageProps}>
    {component}
  </IntlProviderWrapper>
);

export const IntlProviderWrapper = ({
  children,
  pageProps,
  locale,
}: {
  children: JSX.Element;
  pageProps: TranslationsType;
  locale: I18NLocale;
}) => {
  const messages = useMemo(() => {
    return {
      ...getTranslation(locale),
      ...flatData(pageProps[locale] || {}),
    };
  }, [locale, pageProps]);

  return (
    <IntlProvider locale={`${locale}`} messages={messages}>
      {children}
    </IntlProvider>
  );
};
