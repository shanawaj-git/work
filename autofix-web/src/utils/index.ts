import { itemType } from "@/containers/RecommendedMaintenance";
import {
  AssetType,
  fetchCommonData,
  FooterType,
  getAssets,
  getAssetsData,
  getLabelData,
  getLabels,
  HomeActionType,
  LabelType,
  ServiceType,
} from "@/services/common";
import { TestimonialType } from "@/services/home/testimonial";

export const utils = "UTILS";
const GRAPHQL_CLIENT_READ_TIMEOUT = 5000;

interface getCommonPageContentParams {
  locale: I18NLocale;
  pageName: string;
}

export interface HomeContentType {
  labels?: LabelType[];
  assets?: AssetType[];
  services?: ServiceType[];
  actions?: HomeActionType[];
  testimonials?: TestimonialType[];
  footer?: FooterType;
}

export type I18NLocale = "en" | "ar";

/**
 * Will Return a Data object with Labels and Assets Regarding the page from Strapi
 * @param  locale , pageName
 * @returns Object with Labels and Assets
 */

export const getCommonPageContent = async ({
  locale,
  pageName,
}: getCommonPageContentParams): Promise<HomeContentType> => {
  let data: HomeContentType = {};
  return new Promise(async (resolve, reject) => {
    try {
      const commonData = await (
        await fetchCommonData(locale as I18NLocale, pageName)
      ).data;
      const assets = getAssets(commonData);
      const assetsData = getAssetsData(assets);
      const labels = getLabelData(getLabels(commonData));
      data = { assets: assetsData, labels };
      resolve(data);
    } catch (error) {
      resolve(data);
    }
  });
};

interface rejectEarlyType<T> {
  fn: () => Promise<T>;
  def: T;
  timeout?: number;
}

export const rejectEarly = <T>({
  fn,
  def,
  timeout = GRAPHQL_CLIENT_READ_TIMEOUT,
}: rejectEarlyType<T>): Promise<T> => {
  return new Promise(async (resolve, _reject) => {
    setTimeout(() => {
      resolve(def);
    }, timeout);
    try {
      const out: T = await fn();
      resolve(out);
    } catch (error) {
      resolve(def);
    }
  });
};

export const validateEmail = (email: string) =>
  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);

export const flatData = (obj: { [key: string]: {} }) =>
  Object.values(obj).reduce((acc: {}, curr: {}) => {
    return { ...acc, ...curr };
  }, {});

export const buildImageUrl = (path: string): string =>
  process.env.NEXT_PUBLIC_STRAPI_SERVICE?.concat(path) || "";

export const groupBy = <T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => string
) =>
  array.reduce((acc, value, index, array) => {
    (acc[predicate(value, index, array)] ||= []).push(value);
    return acc;
  }, {} as { [key: string]: T[] });

export const includesCaseInsensitive = (
  string: string,
  substring: string
): Boolean => {
  return string.toLowerCase().includes(substring.toLowerCase());
};
export const getTotalForMax = (data: itemType[]) => {
  return data.reduce((acc: number, curr: { cost: string }) => {
    let value = acc;
    value += +parseFloat(
      (curr.cost.match(/[0-9]+/gm) || ["0"]).splice(-1)[0]
    ).toFixed(2);
    return value;
  }, 0);
};

export const getCurrency = (data: itemType[]) => {
  return (data[0].cost.match(/[A-Z]+/gm) || ["USD"]).splice(-1)[0];
};

export const getCurrentLocation = () => {
  return new Promise((resolve: (e: GeolocationCoordinates) => void, reject) => {
    if (!navigator) throw new Error("Navigator doesn't exist");
    if (!navigator.geolocation) throw new Error("Geolocation doesn't exist");
    navigator.geolocation.getCurrentPosition(
      (position: GeolocationPosition) => {
        return resolve(position.coords);
      },
      reject,
      {
        maximumAge: 5000,
        enableHighAccuracy: true,
      }
    );
  });
};

interface getCurrentLiveLocationType {
  callBack: (e: GeolocationPosition) => void;
  onError: (e: GeolocationPositionError | Error) => void;
}

export const getCurrentLiveLocation = ({
  callBack,
  onError,
}: getCurrentLiveLocationType) => {
  if (!navigator) throw new Error("Navigator doesn't exist ");
  if (!navigator.geolocation) throw new Error("Geolocation doesn't exist ");
  navigator.geolocation.watchPosition((position: GeolocationPosition) => {
    callBack(position);
  }, onError);
};
