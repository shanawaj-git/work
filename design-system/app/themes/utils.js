import { getTheme } from './getTheme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
  focus: `0px 0px 10px ${getTheme().theme.color.d100}`,
};

const boxShadowSelector = level => BOX_SHADOW_MAPPING[level];
const colorSelector = color => getTheme().theme.color[color.toLowerCase()];
export { boxShadowSelector, colorSelector };
