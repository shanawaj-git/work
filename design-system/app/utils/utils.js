import { v4 } from 'uuid';
import _debounce from 'lodash/debounce';
/**
 * Helper: Convert Array of coordinates to an object of cordinates
 * @param {*} cordinates : Array [lat,lng]
 * @returns
 */
export const transformedCordinates = cordinates => {
  const [lat, lng] = cordinates;
  return {
    lat,
    lng,
  };
};

export const readableCordinates = ({ lat, lng }) => ({
  lat: lat(),
  lng: lng(),
});

export const uuid = () => v4();

const MAP_ADDRESS_COMPONENT_TYPE = {
  LOCALITY: 'locality',
  COUNTRY: 'country',
};

export const getCityAndCountry = address => {
  let city;
  let country;
  // eslint-disable-next-line no-unused-expressions
  address?.forEach(component => {
    const { types } = component;
    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.LOCALITY) > -1) {
      city = component.long_name;
    }

    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.COUNTRY) > -1) {
      country = component.long_name;
    }
  });
  return {
    city,
    country,
  };
};

export const debounce = (func, wait) => _debounce(func, wait);

export const canBeRenderedInTypographyComponent = obj =>
  typeof obj === 'string' ||
  obj?.type?.displayName === 'MemoizedFormattedMessage';
