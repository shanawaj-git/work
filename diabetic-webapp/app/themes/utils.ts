
import { diabeticTheme } from './theme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
};

const boxShadowSelector = (level: string): string => BOX_SHADOW_MAPPING[level];

const colorSelector = (color: string): string =>
  diabeticTheme.color[color.toLowerCase()];

export { boxShadowSelector, colorSelector };
