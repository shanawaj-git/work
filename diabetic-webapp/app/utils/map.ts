export const getAddress = (mapOutput): Object => {
  const {
    form: { direction, flatvillaname, flatvillano },
    addressFetched: {
      formatted_address,
      geometry: {
        location: { lat, lng },
      },
    },
  } = mapOutput;
  return {
    flatVillaNumber: flatvillano,
    formattedText: formatted_address,
    buildingName: flatvillaname,
    userNotes: direction,
    latitude: lat,
    longitude: lng,
  };
};
