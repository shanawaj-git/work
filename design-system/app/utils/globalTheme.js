// eslint-disable-next-line import/no-cycle
import { getTheme } from 'themes/getTheme';
import { SETTINGS } from '../config/organizations';

let currentTheme = 'DEFAULT';
const newThemes = {};

const validateTheme = type => {
  if (!SETTINGS[type])
    throw new Error('ERROR : Theme name provided is not defined');
};

class Theme {
  getInstance() {
    return this;
  }

  getTheme() {
    return currentTheme;
  }

  getDynamicThemes() {
    return newThemes;
  }

  setTheme(theme) {
    if (typeof theme === 'string') {
      validateTheme(theme);
      currentTheme = theme;
    } else {
      newThemes[theme.name] = theme;
      currentTheme = theme.name;
    }
  }

  getActiveTheme() {
    return getTheme();
  }
}
export const GlobalTheme = new Theme();
