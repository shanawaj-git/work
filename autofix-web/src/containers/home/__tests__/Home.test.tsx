import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "../index";
import { withWrapper } from "@/utils/IntlProviderWrapper";

describe("<Home />", () => {
  test("it should", () => expect(true).toBeTruthy());

  test("it should mount", () => {
    const tree = render(withWrapper(<Home />, { en: {} }));
    tree.rerender(withWrapper(<Home />, { ar: {} }));

    expect(tree.container.innerHTML).toMatchSnapshot();
  });
});
