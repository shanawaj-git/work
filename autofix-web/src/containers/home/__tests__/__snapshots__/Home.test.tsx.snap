// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> it should mount 1`] = `
"<div data-testid=\\"main\\" class=\\"min-h-screen \\"><div data-testid=\\"Welcome\\" style=\\"background-image: url(/);\\" class=\\"min-h-screen flex flex-col justify-between px-4 pt-8 pb-8 text-white bg-black bg-no-repeat bg-cover\\"><div class=\\"flex justify-between\\"><span style=\\"box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;\\"><span style=\\"box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;\\"><img style=\\"display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;\\" alt=\\"\\" aria-hidden=\\"true\\" src=\\"data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2786%27%20height=%2724%27/%3e\\"></span><img alt=\\"logo\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" data-nimg=\\"intrinsic\\" style=\\"position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;\\"><noscript></noscript></span><button class=\\"root StyledButton-sc-1apep4b-0 fcMYug MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"><img src=\\"menu-icon.svg\\" alt=\\"Menu\\"></span><div style=\\"visibility: visible;\\"><span class=\\"MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root\\">Menu</span></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><div><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-5xl css-x61q1w-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Car service,
made simple</span></p><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\" style=\\"border-color: #e25f6d;\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 text-base css-x61q1w-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> Fixing your car can be stressful, and it’s difficult to know if you’re being ripped off. With Autofix you can quickly find a trusted mechanic in your local area, compare prices and reviews, then book a meeting.</span></p></div><div class=\\"flex mt-7\\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"home-welcome-button-start\\"><div style=\\"visibility: visible;\\"><p class=\\"MuiTypography-root MuiTypography-body2 css-ry76j1-MuiTypography-root\\">Start Now</p></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div class=\\"w-3\\"></div><button class=\\"root StyledButton-sc-1apep4b-0 cDZhTx !border-[0.2px] MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-xsqe0f-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\"><p class=\\"MuiTypography-root MuiTypography-body2 text-white css-ry76j1-MuiTypography-root\\">User rating</p></div><span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon\\"><div class=\\"flex justify-center\\"><img src=\\"star.svg\\" alt=\\"\\"><p class=\\"MuiTypography-root MuiTypography-body2 pt-1 pl-1 text-white css-ry76j1-MuiTypography-root\\">4.8</p></div></span><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div><div data-testid=\\"intro-section\\" class=\\"min-h-screen flex flex-col justify-between py-[100px] text-white bg-cover bg-no-repeat\\"><div class=\\"px-6\\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-4xl whitespace-pre-line css-x61q1w-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> All services, all car brands</span></p><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\" style=\\"border-color: #e25f6d;\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 text-base css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> We offer all kinds of services for all car brands and make sure that your car is always in good hands. Book an appointment today!</span></p></div></div><div class=\\"flex flex-col w-full my-10 space-y-5 overflow-auto\\"><div class=\\"flex items-center w-full space-x-5 overflow-none \\"></div><div class=\\"flex items-center w-full space-x-5 overflow-none \\"></div><div class=\\"flex items-center w-full space-x-5 overflow-none \\"></div></div><div class=\\"w-full px-6\\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">How can we help?</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div data-testid=\\"howto-section\\" class=\\"min-h-screen flex flex-col justify-between py-[100px] space-y-8\\"><div class=\\"px-6 \\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 leading-10 text-[34px]  css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> We are here to make 
your car service easy &amp; 
 transparent</span></p><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\" style=\\"border-color: #e25f6d;\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 text-base font-light css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> Enjoy a transparent, easy process with no 
 hidden fees or catches and get affordable 
 car servicing at our nationwide network of 
 independent garages.</span></p></div></div><div class=\\"flex flex-col items-center justify-center w-full px-6 space-y-4\\"></div></div><div data-testid=\\"testimonial-section\\" class=\\"min-h-screen flex flex-col justify-between py-[100px] text-white bg-cover bg-no-repeat\\"><div class=\\"flex flex-col flex-1 \\"><div class=\\"px-6\\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-4xl whitespace-pre-line css-x61q1w-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Trusted
by customers</span></p><div data-testid=\\"article-component-divider\\" style=\\"border-color: #e25f6d;\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] mt-2\\"></div><div color=\\"white\\" class=\\"text-4xl whitespace-pre-line\\"></div></div></div><div class=\\"sc-d6b1016c-0 jvznlT carousel h-full\\"><div style=\\"overflow-x: hidden; flex: 1; flex-direction: flex; height: 100%;\\"><div style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\" class=\\"react-swipeable-view-container\\"><div style=\\"width: 100%; flex-shrink: 0; overflow: auto;\\" aria-hidden=\\"false\\" data-swipeable=\\"true\\"><div><img class=\\"MuiBox-root css-ijk13v\\" src=\\"https://images.unsplash.com/photo-1537944434965-cf4679d1a598?auto=format&amp;fit=crop&amp;w=400&amp;h=250&amp;q=60\\" alt=\\"San Francisco – Oakland Bay Bridge, United States\\"></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic css-rh92k-MuiPaper-root-MuiMobileStepper-root\\"><div class=\\"MuiMobileStepper-dots css-1384nzf-MuiMobileStepper-dots\\"></div><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"next-btn\\"><div style=\\"visibility: visible;\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"ArrowForwardIcon\\"><path d=\\"m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z\\"></path></svg></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div><div data-testid=\\"ToFix\\" class=\\"py-24 px-6\\"><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-5xl css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> We’re here to help you fix your car</span></p><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] my-4\\"></div><div class=\\"text-5xl\\"><span class=\\"text-base \\">If your car is in need of some love, we’re here to help. Just tell us what needs fixing, and we’ll book it into one of our approved workshops. It’s that easy!</span></div></div><div class=\\" pt-7 flex \\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup !text-[14px] MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"home-tofix-startnow\\"><div style=\\"visibility: visible;\\">Start Now</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div class=\\"w-3\\"></div><button class=\\"root StyledButton-sc-1apep4b-0 cDZhTx !border-[0.2px] !text-[14px] MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-xsqe0f-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">Learn about my car</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div data-testid=\\"SignUp\\" class=\\" text-white py-24 py-100 px-6\\"><div data-testid=\\"article-component\\"><h1 class=\\"MuiTypography-root MuiTypography-h1  !font-normal !tracking-normal css-1yzmk4e-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Get a free booking!</span></h1><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] my-4\\" style=\\"border-color: #e25f6d;\\"></div><div color=\\"white\\" class=\\" !font-normal !tracking-normal\\" typographytype=\\"largeTitle\\"><span class=\\"text-base \\">Subscribe to receive our offers and enjoy 1 booking for free! No spam, we promise.</span></div></div><div class=\\"root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF sc-edb9ce2-0 gBVPiv mt-4  css-1nrlq1o-MuiFormControl-root\\"><div class=\\"MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root\\" data-testid=\\"email-signup-input\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1c2i806-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"true\\" for=\\":r0:\\" id=\\":r0:-label\\"><span class=\\"text-white\\">E-mail</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"\\" id=\\":r0:\\" name=\\"\\" placeholder=\\"Your email\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\"></div></div></div><button class=\\"root StyledButton-sc-1apep4b-0 jVvup my-7 !normal-case MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">Sign me up</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><p class=\\"MuiTypography-root MuiTypography-body1 !font-light opacity-80 text-sm css-x61q1w-MuiTypography-root\\">We wont pass your details on anyone else. By clicking sign me up button you agree to our <span style=\\"color: rgb(226, 95, 109);\\"> terms of use </span> and<span style=\\"color: rgb(226, 95, 109);\\"> privacy policy </span>.</p></div><div data-testid=\\"Section\\" class=\\"min-h-screen flex flex-col text-white\\"><div class=\\"px-6 py-20 flex-1\\"><span style=\\"box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;\\"><span style=\\"box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;\\"><img style=\\"display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;\\" alt=\\"\\" aria-hidden=\\"true\\" src=\\"data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2786%27%20height=%2724%27/%3e\\"></span><img alt=\\"logo\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" data-nimg=\\"intrinsic\\" style=\\"position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;\\"><noscript></noscript></span></div><div class=\\"flex flex-1 justify-center items-center border-t border-warmGray-600\\"><p class=\\"MuiTypography-root MuiTypography-body2 mr-4 opacity-80 css-ry76j1-MuiTypography-root\\"></p><span style=\\"box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;\\"><span style=\\"box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;\\"><img style=\\"display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;\\" alt=\\"\\" aria-hidden=\\"true\\" src=\\"data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27129%27%20height=%2723%27/%3e\\"></span><img alt=\\"footer-logo\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" data-nimg=\\"intrinsic\\" style=\\"position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;\\"><noscript></noscript></span></div></div></div>"
`;
